File: test-support/src/main/java/com/rafaelfiume/prosciutto/test/DependsOnServerRunningRule.java
package com.rafaelfiume.prosciutto.test;

import org.junit.rules.ExternalResource;

public class DependsOnServerRunningRule extends ExternalResource {

    private final StubbedServer server = new StubbedServer();

    public void primeSuccessfulResponse(String contextPath, String response) {
        server.primeSuccessfulResponse(contextPath, response);
    }

    @Override
    protected void before() throws Throwable {
        server.start();
    }

    @Override
    protected void after() {
        try {
            server.stop();
        } catch (Exception e) {
            throw new RuntimeException("failed stopping server", e);
        }
    }

}

File: test-support/src/main/java/com/rafaelfiume/prosciutto/test/DependsOnServerRunningRule.kt
package com.rafaelfiume.prosciutto.test

import org.junit.rules.ExternalResource

class DependsOnServerRunningRule : ExternalResource() {

    private val server = StubbedServer()

    fun primeSuccessfulResponse(contextPath: String, response: String) {
        server.primeSuccessfulResponse(contextPath, response)
    }

    fun primeServerErrorWhenRequesting(contextPath: String) {
        server.primeServerErrorWhenRequesting(contextPath)
    }

    @Throws(Exception::class)
    fun stop() {
        server.stop()
    }

    @Throws(Throwable::class)
    override fun before() {
        server.start()
    }

    override fun after() {
        try {
            server.stop()
        } catch (e: Exception) {
            throw RuntimeException("failed stopping server", e)
        }

    }

}
